File: mono/src/api/Analyser.cs
using Sentan.Core;

namespace Sentimentalizer.Api
{
    public static class Analyser
    {
        static Corpus _positiveCorpuse = new Corpus();
        static Corpus _negativeCorpus = new Corpus();

        public static ClassificationResult Analyse(string text)
        {
            return new Classifier(_positiveCorpuse, _negativeCorpus).ClassifyData(text);
        }

        public static void TrainPositive(string path)
        {
            _positiveCorpuse.LoadFromDirectory(path);
        }

        public static void TrainNegative(string path)
        {
            _negativeCorpus.LoadFromDirectory(path);
        }
    }
}

File: rb/src/api/analyser.rb
require "../engine/corpus"
require "../engine/classifier"

class Analyser

  @@positive = Corpus.new
  @@negative = Corpus.new

  def Analyser.train_positive path
    @@positive.load_from_directory path
  end

  def Analyser.train_negative path
    @@negative.load_from_directory path
  end

  def Analyser.analyse sentence
    Classifier.new(@@positive, @@negative).classify(sentence)
  end
end
